internal/host/validator.go

```diff
@@ -372,6 +398,30 @@ func (v *validator) diskEncryptionRequirementsSatisfied(c *validationContext) Va
 		return ValidationPending
 	}
 
+	if hostutil.IsDay2Host(c.host) {
+		luks, err := v.getDiskEncryptionForDay2(c.host)
+		if err != nil {
+			return ValidationPending
+		}
+		if luks == nil {
+			// Disk encryption is disabled for workers on day1 cluster
+			return ValidationSuccessSuppressOutput
+		}
+
+		c.cluster.DiskEncryption = &models.DiskEncryption{}
+		if swag.BoolValue(luks.Clevis.Tpm2) {
+			// If Tpm2 is enabled for workers, check whether supported by the host.
+			c.cluster.DiskEncryption.Mode = swag.String(models.DiskEncryptionModeTpmv2)
+			return boolValue(c.inventory.TpmVersion == models.InventoryTpmVersionNr20)
+		} else if len(luks.Clevis.Tang) != 0 {
+			c.cluster.DiskEncryption.Mode = swag.String(models.DiskEncryptionModeTang)
+			return ValidationSuccess
```

Comment:
I think we can return `ValidationSuccessSuppressOutput` here as we didn't do any actual validation.